Mathlib/Analysis/NormedSpace/OperatorNorm.lean

```diff
@@ -981,22 +971,9 @@ def prodMapL : (M₁ →L[𝕜] M₂) × (M₃ →L[𝕜] M₄) →L[𝕜] M₁
       apply funext
       rintro ⟨φ, ψ⟩
       refine' ContinuousLinearMap.ext fun ⟨x₁, x₂⟩ => _
-      -- Porting note: mathport suggested:
-      -- ```
-      -- simp only [add_apply, coe_comp', coe_fst', Function.comp_apply, compL_apply, flip_apply,
-      --   coe_snd', inl_apply, inr_apply, Prod.mk_add_mk, add_zero, zero_add, coe_prodMap'
-      --   Prod_map, Prod.mk.inj_iff, eq_self_iff_true, and_self_iff]
-      -- rfl
-      -- ```
-      -- Frustratingly, in `mathlib3` we can use:
-      -- ```
-      -- dsimp   -- ⊢ (⇑φ x.fst, ⇑ψ x.snd) = (⇑φ x.fst + 0, 0 + ⇑ψ x.snd)
```

Comment:
Does dsimp work with the same trick? I think this proof is easier to maintain if so

Reply:
Sorry I am confused as to what is being asked here. Swap out `dsimp` for `simp`?

Reply:
I pushed a `simp? says`. Let me know if you think that is an improvement in this regard. (It also told me I could drop some parens).

Reply:
The lean3 proof was apparently `dsimp; simp`. I'm suggesting you restore that proof as `dsimp [(foo)]; simp [(bar)]` as appropriate, ideally reaching the same intermediate goal state.

Reply:
I tried that at first. Let me try again.

Reply:
Best I can do is 
```lean
dsimp -- does some things 
simp [(add_apply), (comp_apply)]
```
in Lean 3 the `add_apply` and `comp_apply` get called with the `dsimp` leaving only `zero_add` and `add_zero` to clean up but the parens trick doesn't work here with `dsimp`. 

We could also just do 
```lean
simp [(add_apply), (comp_apply)]
```
alone. Thoughts?

Reply:
I see that your PR about `Algebra.id` was merged recently. Is this workaround still needed?

Reply:
Unfortunately, yes.

Reply:
I found the declarations that need to be reducible.